src/components/atoms/input.tsx

```diff
@@ -1,20 +1,40 @@
-import React from 'react'
+import { Eye, EyeOff } from 'lucide-react'
+import React, { useState } from 'react'
 
 interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
   className?: string
   placeholder?: string
+  isPassword?: boolean
 }
 
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, placeholder, ...props }, ref) => {
+  ({ className, placeholder, isPassword = false, ...props }, ref) => {
+    const [showPassword, setShowPassword] = useState(false)
+
+    let inputType = 'text';
+    if (isPassword) {
+      inputType = showPassword ? 'text' : 'password';
+    }
+
     return (
-      <input
-        type="text"
-        ref={ref}
-        className={`rounded border p-2 ${className}`}
-        placeholder={placeholder}
-        {...props}
-      />
+      <div className="relative w-full">
+        <input
+          type={inputType}
+          ref={ref}
+          className={`rounded border p-2 ${className} ${isPassword ? 'pr-10' : ''}`}
+          placeholder={placeholder}
+          {...props}
+        />
+        {isPassword && (
+          <button
+            type="button"
+            onClick={() => setShowPassword(!showPassword)}
+            className="absolute right-2 top-1/2 transform -translate-y-1/2 focus:outline-none"
+          >
+            {showPassword ? <EyeOff className="h-5 w-5" /> : <Eye className="h-5 w-5" />}
+          </button>
+        )}
+      </div>
```

Comment:
I have implemented the password hidden mechanism to the abstract component (Input) in my [PR](https://github.com/kg-procurement/kg-procurement-ui/pull/25). We can change it to that later in the integration PR because the my PR isn't merged yet.

Reply:
Hey, I finally merged @edutjie's PR, so please kindly merge his features and use his implementation of toggleable hidden input for password

CC: @bryannaufal